templates/commands/render.go

```diff
@@ -367,76 +366,123 @@ func parseAndExecuteGoTmpl(m model.String, inputs map[string]string) (string, er
 }
 
 // "from" may be a file or directory. "pos" is only used for error messages.
-func copyRecursive(pos *model.ConfigPos, from, to string, rfs renderFS, overwrite bool) error {
-	return fs.WalkDir(rfs, from, func(path string, d fs.DirEntry, err error) error { //nolint:wrapcheck
+func copyRecursive(pos *model.ConfigPos, fromRoot, toRoot string, rfs renderFS, overwrite, dryRun bool) error {
+	return fs.WalkDir(rfs, fromRoot, func(path string, de fs.DirEntry, err error) error { //nolint:wrapcheck
 		if err != nil {
 			return err // There was some filesystem error. Give up.
 		}
 
 		// We don't have to worry about symlinks here because we passed
 		// DisableSymlinks=true to go-getter.
 
-		relToSrc, err := filepath.Rel(from, path)
-		if err != nil {
-			return model.ErrWithPos(pos, "filepath.Rel(%s,%s): %w", from, path, err) //nolint:wrapcheck
+		if de.IsDir() {
+			return copyRecursiveDir(path, de, pos, fromRoot, toRoot, rfs, dryRun)
 		}
 
-		dst := filepath.Join(to, relToSrc)
+		return copyRecursiveFile(path, de, pos, fromRoot, toRoot, rfs, overwrite, dryRun)
+	})
+}
 
-		// The spec file may specify a file to copy that's deep in a
-		// directory tree, without naming its parent directory. We can't
-		// rely on WalkDir having traversed the parent directory of $path,
-		// so we must create the target directory if it doesn't exist.
-		dirToCreate := dst
-		if !d.IsDir() {
-			dirToCreate = filepath.Dir(dst)
-		}
+// A helper function that's called by the WalkDirFunc inside copyRecursive in
+// the case that the source path is a directory.
+func copyRecursiveDir(path string, de fs.DirEntry, pos *model.ConfigPos, fromRoot, toRoot string, rfs renderFS, dryRun bool) error {
+	relToSrc, err := filepath.Rel(fromRoot, path)
+	if err != nil {
+		return model.ErrWithPos(pos, "filepath.Rel(%s,%s): %w", fromRoot, path, err) //nolint:wrapcheck
+	}
+	dst := filepath.Join(toRoot, relToSrc)
```

Comment:
Tabs seem to be off on these lines.

Reply:
Can you be more specific? It looks fine to me, and "gofumpt" is fine with it.